Design_evaluation (#19)
chhoumann/p4: Report/sections/2_design/Design_evaluation.tex

\section{Design evaluation} \label{chap:design evaluation}

> What is significant about developing a language is to carefully consider various parameters that could be around the language itself and the setting up of essential criteria for the programming language. 

Reviewer: ```suggestion
What is significant about developing a language, is to carefully consider various parameters concerning the language itself and setting up essential criteria for the programming language. 
```

> Therefore, it is crucial to be able to evaluate the different parameters using different tools. Thus, the group has chosen to use an already set up table, which covers all the most important criteria for setting up a programming language. 

Reviewer: omformuler "use an already set up table".

Reviewer: burde man vide hvad table er?

> Therefore, it is crucial to be able to evaluate the different parameters using different tools. Thus, the group has chosen to use an already set up table, which covers all the most important criteria for setting up a programming language. 

Reviewer: tror ikke helt det er underforstået hvad 'table' er.

Måske skrive noget ala 'use an already existing set of critera...'

It has been selected as the need to find other parameters seemed unnecessary, as this table contains all the critical parameters behind the coding language to be produced.

\fig{0.8}{Design_Criteria}{Table about design criteria }

> The simplicity is to be able to keep the language simple to read and understand for everyone. It helps to keep the learning curve at a superficial level.

Reviewer: ```suggestion
Simplicity is about keeping the language simple to read and understand for everyone. It helps to keep the learning curve at a superficial level.
```

> Orthogonality is about how many different ways to put together data structures. High orthogonality describes that the data structure of the language can be composed in many different ways.

Reviewer: ```suggestion
Orthogonality is about how many different ways data structures can be put together. High orthogonality describes that the data structure of the language can be composed in many different ways.
```

Data types are the different types of data that are possible to work within the language. By simplifying the data type, names in the language can help the beginner understand the various properties behind the language.

Syntax design is a broad concept within the language, which can be the actual use of unique and reserved words.

> Abstraction support is the precise handling of the code's reuse in the language itself without repeating the code writing.

Reviewer: Synes sætningen på linje 19 bør omformuleres

måske mest ift. 'the code's reuse in the language itself'

> Expressiveness is the very meaning of ways one can write down different operations in the language. Being able to make them simple to write down helps make the language and readability better.

Reviewer: ```suggestion
Expressiveness is the very meaning of ways one can write down different operations in the language. Being able to make them simple to write down helps make the language better and more readable.
```

> Type checks are used to be able to test type errors that could appear in the program. This can be done by either the compiler intercepts it or when the program is running. There can be many different type checks in a programming language. It helps that you quickly have a chance to detect faults in good time and solve them quickly.

Reviewer: ```suggestion
Type checks are used to be able to test type errors that could appear in the program. This can be done by either the the compiler intercepting it or when the program is running. There can be many different type checks in a programming language. It helps detecting faults in good time and solve them quickly.
```

> Exception handling helps to detect errors that could occur while running the program. Besides, it provides a good overview of how mistakes could have happened, and it helps to find out how they can be resolved.

Reviewer: ```suggestion
Exception handling helps to detect errors that could occur while running the program. Additionally, it provides a good overview of how mistakes could have happened, and it helps to find out how they can be resolved.
```

Reviewer: Var det ikke det her Gigi kommenterede? 
At den ikke 'detectede' errors men håndterede dem.

Aliasing comes from naming several different names, which you then have access to at the same memory cell.

\subsection{Readability}

> Readability is one of the most important for the language itself, as it is intended for beginners in programming. 

Reviewer: ```suggestion
Readability is one of the most important criteria for the language itself, as it is intended for beginners in programming. 
```

Therefore, the language must be set up, and only the most valuable data types have been used to set it up. It is also important for the language itself that the syntax setup is well thought out to create minor confusion for beginners.
 This is done by keeping the orthogonality low, and the special words will be turned into reserved words to increase readability. 

> Setting specific words as reserved words helps define particular designs in the program, allowing the language to be easily readable.

Reviewer: ```suggestion
Setting specific words as reserved words helps define particular designs in the program, allowing the language to be more readable.
```
